src/Cache/Cache.php

```diff
@@ -158,32 +158,8 @@ protected static function _buildEngine(string $name): void
         try {
             $registry->load($name, $config);
         } catch (RuntimeException $e) {
-            if (!array_key_exists('fallback', $config)) {
-                $registry->set($name, new NullEngine());
-                trigger_error($e->getMessage(), E_USER_WARNING);
-
-                return;
-            }
-
-            if ($config['fallback'] === false) {
-                throw $e;
-            }
-
-            if ($config['fallback'] === $name) {
-                throw new InvalidArgumentException(sprintf(
-                    '"%s" cache configuration cannot fallback to itself.',
-                    $name
-                ), 0, $e);
-            }
-
-            /** @var \Cake\Cache\CacheEngine $fallbackEngine */
-            $fallbackEngine = clone static::pool($config['fallback']);
-            $newConfig = $config + ['groups' => [], 'prefix' => null];
-            $fallbackEngine->setConfig('groups', $newConfig['groups'], false);
-            if ($newConfig['prefix']) {
-                $fallbackEngine->setConfig('prefix', $newConfig['prefix'], false);
-            }
-            $registry->set($name, $fallbackEngine);
+            $fallbackEngine = static::getFallbackEngine($name, $config, $e);
+            static::registerFallbackEngine($fallbackEngine, $name, $config);
```

Comment:
Could these methods be combined to help reduce the new surface area we're adding?

Reply:
Thanks for your comments!

Maybe it is possible. 

I initially created two because of DebugEngine gets another DebugEngine when calling `getFallbackEngine` so it gets `$fallbackEngine->engine()` instead of using it directly as Cache class does. Otherwise we would end with a DebugEngine inside a DebugEngine.